spec/javascripts/accordion-with-descriptions-spec.js

```diff
@@ -0,0 +1,181 @@
+    expect($element).toHaveClass("js-accordion-with-descriptions");
+  });
+
+  // Remove the class .js-hidden
+  it("does not have a class of js-hidden", function () {
+    expect($element).not.toHaveClass("js-hidden");
+  });
+
+  // Add a subsection controls div, with a class of .js-subsection controls
+  it("has a child element with a class of subsection-controls", function () {
+    expect($element).toContain('.js-subsection-controls');
+  });
+
+  // Insert a button inside .js-subsection-controls
+  it("has a child element which is a button", function () {
+    expect($element).toContain('.js-subsection-controls button');
+  });
+
+  // Set the aria-expanded attribute to false, as all subsections are initially closed
+  it("has an open/close all button with an aria-expanded attribute and it is false", function () {
+    var $button = $element.find('.js-subsection-controls button');
+
+    expect($button).toHaveAttr("aria-expanded", "false");
+  });
+
+  // Set the text inside the button to "Open all"
+  it("has an open/close all button with text inside which is equal to Open all", function () {
+    var $openCloseAllButton = $element.find('.js-subsection-controls button');
+
+    expect($openCloseAllButton).toHaveText("Open all");
+  });
+
+  // Ensure that the value for the aria-controls attribute matches the IDs of all of the subsections
+  it("has a value for the aria-controls attribute which includes all of the subsection_content_IDs", function () {
+    var $openCloseAllButton = $element.find('.js-subsection-controls button');
+
+    expect($openCloseAllButton).toHaveAttr('aria-controls','subsection_content_0 subsection_content_1 ');
+  });
+
+  // Insert a button into each subsection title
+  // Set the correct text and aria attributes (aria-expanded, aria-controls) for the button
+  it("has a h2 with a class of .subsection__title with a child element which is a button", function () {
+    var $subsectionButton = $element.find('.subsection__title button:first');
+
+    expect($subsectionButton).toHaveClass('subsection__button');
+    expect($subsectionButton).toHaveAttr('aria-expanded','false');
+    expect($subsectionButton).toHaveAttr('aria-controls','subsection_content_0');
+  });
+
+  // Ensure the wrapper for the list of links is initially hidden
+  it("has two subsection-content items which are initially hidden", function () {
+    var $subsectionContent = $element.find('.subsection__content');
+
+    expect($subsectionContent).toHaveLength(2);
+    expect($subsectionContent).toHaveClass('js-hidden');
+  });
+
+  // Ensure that the subsection-icon div has been inserted
+  it("has a header with a child element which has a class of .subsection-icon", function () {
+    var $subsectionHeader = $element.find('.subsection__header');
+
+    expect($subsectionHeader).toContain('.subsection__icon');
+  });
+
+  // When all sections are closed, make sure there are no .is-open classes
```

Comment:
I like all the explanation of what is going on. I think this type of comment can work well as another `describe` block so it becomes a little more "alive" and more likely to stay updated. Also, when it's a describe/context block it annotates the test runner output which tells a more complete story which is nice (sometimes).